### PR TITLE
Scaffolding for datastore package

### PR DESCRIPTION
**Pull request by Amplifiyer**
I haven't touched any codegen source files since they will be updated with real implementation.

This PR is purely to get the scaffolding out so real code changes next time onward are more easily visible.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by kjones** [comment]
Just a quick fly-by comment on naming consistency. All other platforms (js, ios, android) all use `DataStore` naming as opposed to `Datastore`.

File: packages/amplify_datastore/LICENSE [hunk 1]
```diff
@@ -0,0 +1,176 @@
+TODO: Add your license here.
```
Comment (haverchuck) [inline packages/amplify_datastore/LICENSE:1]: Can we remove this TODO line?

File: packages/amplify_datastore/android/src/main/kotlin/com/amazonaws/amplify/amplify_datastore/AmplifyDataStorePlugin.kt [hunk 2]
```diff
@@ -0,0 +1,36 @@
+package com.amazonaws.amplify.amplify_datastore
```
Comment (haverchuck) [inline packages/amplify_datastore/android/src/main/kotlin/com/amazonaws/amplify/amplify_datastore/AmplifyDataStorePlugin.kt:1]: Don't forget to add license boilerplate.

File: packages/amplify_datastore/example/ios/Podfile [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+require File.expand_path(File.join('packages', 'flutter_tools', 'bin', 'podhelper'), flutter_root)
+
+flutter_ios_podfile_setup
+
+target 'Runner' do
+  use_frameworks!
+  use_modular_headers!
+
+  flutter_install_all_ios_pods File.dirname(File.realpath(__FILE__))
```
Comment (haverchuck) [inline packages/amplify_datastore/example/ios/Podfile:34]: Just as an FYI - this 'flutter_install_all_ios_pods' command is not available in Podfiles generated from older flutter sdk version, and it is needed when setting up unit tests.  Just something to be aware of.

File: packages/amplify_datastore/ios/amplify_datastore.podspec [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  s.description      = <<-DESC
+The DataStore module for Amplify Flutter.
+                       DESC
+  s.homepage = 'https://github.com/aws-amplify/amplify-flutter'
+  s.license = 'Apache License, Version 2.0'
+  s.author = { 'Amazon Web Services' => 'amazonwebservices' }
+  s.source = { :git => 'https://github.com/aws-amplify/amplify-flutter.git' }
+  s.source_files = 'Classes/**/*'
+  s.dependency 'Flutter'
+  s.platform = :ios, '8.0'
```
Comment (haverchuck) [inline packages/amplify_datastore/ios/amplify_datastore.podspec:18]: Now might be a good time to change s.platform to 13 (which I believe is the min version required)?

File: packages/amplify_datastore_plugin_interface/LICENSE [hunk 5]
```diff
@@ -0,0 +1,174 @@
+Apache License
```
Comment (haverchuck) [inline packages/amplify_datastore_plugin_interface/LICENSE:1]: Correct the indentation here.

**Review by haverchuck — CHANGES_REQUESTED** [review]
Looks great, just a couple of minor things:
1. Remove TODO line in license file.
2. Fix indentation in first line of interface license file.
3. I'd suggest changing the platform version in the ios podspec now, but not totally necessary yet.

File: packages/amplify_datastore/LICENSE (hunk 1 above)
Comment (Amplifiyer) [reply packages/amplify_datastore/LICENSE:1]: Removed.

File: packages/amplify_datastore/android/src/main/kotlin/com/amazonaws/amplify/amplify_datastore/AmplifyDataStorePlugin.kt (hunk 2 above)
Comment (Amplifiyer) [reply packages/amplify_datastore/android/src/main/kotlin/com/amazonaws/amplify/amplify_datastore/AmplifyDataStorePlugin.kt:1]: Added.

File: packages/amplify_datastore/ios/amplify_datastore.podspec (hunk 4 above)
Comment (Amplifiyer) [reply packages/amplify_datastore/ios/amplify_datastore.podspec:18]: changed

File: packages/amplify_datastore_plugin_interface/LICENSE (hunk 5 above)
Comment (Amplifiyer) [reply packages/amplify_datastore_plugin_interface/LICENSE:1]: Good catch. Fixed.

**Review by haverchuck — APPROVED** [review]

**Pull request merged**